apis/dscinitialization/v1/servicemesh_types.go

```diff
@@ -0,0 +1,33 @@
+package v1
+
+import operatorv1 "github.com/openshift/api/operator/v1"
+
+// ServiceMeshSpec configures Service Mesh.
+type ServiceMeshSpec struct {
+	// +kubebuilder:validation:Enum=Managed;Removed
+	// +kubebuilder:default=Removed
+	ManagementState operatorv1.ManagementState `json:"managementState,omitempty"`
+	// Mesh holds configuration of Service Mesh used by Opendatahub.
+	Mesh ControlPlaneSpec `json:"controlPlane,omitempty"`
+}
+
+type ControlPlaneSpec struct {
+	// Name is a name Service Mesh Control Plane. Defaults to "data-science-smcp".
+	// +kubebuilder:default=data-science-smcp
```

Comment:
is this the only option?
if there are more static option would be better to use kubebuilder enum.

Reply:
This is just the default. You can use the name you like better when editing the resource.